src/realm/util/compression.cpp

```diff
@@ -428,16 +444,25 @@ std::error_code decompress_libcompression(NoCopyInputStream& compressed, Span<co
     // The first nibble is compression algorithm (where 8 is DEFLATE), and second
     // nibble is window size. RFC 1950 only allows window size 7, so the first
     // byte must be 0x78.
-    if (read_byte(compressed, compressed_buf) != (uint8_t)Algorithm::Deflate)
-        return error::corrupt_input;
+    compression_algorithm algo;
+    switch (Algorithm(read_byte(compressed, compressed_buf))) {
+        case Algorithm::Deflate:
+            algo = COMPRESSION_ZLIB;
+            break;
+        case Algorithm::LZFSE:
+            algo = COMPRESSION_LZFSE;
+            break;
+        default:
+            return error::corrupt_input;
+    }
     // The second byte has flags. Bit 5 is the only interesting one, which
     // indicates if a custom dictionary was used. We don't support that.
     uint8_t flags = read_byte(compressed, compressed_buf);
-    if (flags & 0b10000)
+    if (flags & 0b100000)
```

Comment:
Was this not bit 5 before or is the comment above wrong?

Reply:
It was initially not bit 5 because I misread the RFC and didn't realize it started from bit 0.